### PR TITLE
Add CI templates

### PR DESCRIPTION
**Pull request by ddimaria**
Note: CI linting is failing (over 900 issues).  We will submit a lint-fix PR once all code is in develop.

- [x] Add the CI workflow templates from other libp2p projects
- [x] Add linting scripts to package.json
- [x] Document in README

refs: https://github.com/libp2p/specs/pull/412

### CODE REVIEW

File: .github/workflows/js-test-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      - uses: actions/setup-node@v2
+        with:
+          node-version: lts/*
+      - uses: ipfs/aegir/actions/cache-node-modules@master
+      - run: npx xvfb-maybe npm run --if-present test:electron-main
+      - uses: codecov/codecov-action@81cd2dc8148241f03f5839d295e000b8f761e378 # v3.1.0
+        with:
+          flags: electron-main
+
+  test-electron-renderer:
```
Comment (ckousik) [inline .github/workflows/js-test-and-release.yml:116]: Is this different from electron? We do not have any UI components.

Comment (ddimaria) [reply .github/workflows/js-test-and-release.yml]: Removed

File: .github/workflows/js-test-and-release.yml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v2
+      with:
+        node-version: lts/*
+    - uses: ipfs/aegir/actions/cache-node-modules@master
+    - run: npm run --if-present lint
+    - run: npm run --if-present dep-check
+
+  test-node:
```
[the remark this answers is not in the document]
Comment (ddimaria) [reply .github/workflows/js-test-and-release.yml]: Removed

**Review by John-LittleBearLabs — APPROVED** [review]
Good luck with the merge conflicts.

**Pull request merged**